R/methods-Links.R

```diff
@@ -32,3 +36,39 @@ setMethod("getWeights", "LinkList", function(object) unlist(lapply(object, funct
 setGeneric("getColors", function(object) standardGeneric("getColors"))
 setMethod("getColors", "LinkList", function(object) unlist(lapply(object, function(x) color(x))))
 
+
+toJSONGeneric <- getGeneric("toJSON", package = "veupathUtils")
+
+#' Convert Link object to JSON
+#' 
+#' Converts a Link object to JSON
+#' @param object A Link object
+#' @param named boolean that declares if names should be included
+#' @export
+setMethod(toJSONGeneric, "Link", function(object, named = c(FALSE, TRUE)) {
+    named <- veupathUtils::matchArg(named)
+    tmp <- character()
+
+    tmp <- paste0('"source":', jsonlite::toJSON(jsonlite::unbox(id(source(object)))))
+    tmp <- paste0(tmp, ',"target":', jsonlite::toJSON(jsonlite::unbox(id(target(object)))))
+    if (!!length(weight(object))) tmp <- paste0(tmp, ',"weight":', jsonlite::toJSON(jsonlite::unbox(weight(object))))
+    if (!!length(color(object))) tmp <- paste0(tmp, ',"color":', jsonlite::toJSON(jsonlite::unbox(color(object))))
+    tmp <- paste0(tmp, ',"isDirected":', jsonlite::toJSON(jsonlite::unbox(isDirected(object))))
+
+    tmp <- paste0('{', tmp, '}')
+    if (named) {
+        tmp <- paste0('{"link":', tmp, '}')
+    }
+
+    return(tmp)
+})
+
+#' @export
+setMethod(toJSONGeneric, signature("LinkList"), function(object, named = c(TRUE, FALSE)) {
+    named <- veupathUtils::matchArg(named) 
+    tmp <- veupathUtils::S4SimpleListToJSON(object, FALSE)
```

Comment:
does this work just fine using the simple list version? If yes do we even need the `Link` method just above?

Reply:
i dont understand the question i think..... but yes, we need the Link method. it gets called by S4SimpleListToJson